src/components/appliedRules/PersonaScore.jsx

```diff
@@ -0,0 +1,268 @@
+import { useEffect, useState } from 'react';
+
+import { hitsConfig } from '@/config/hitsConfig';
+
+const PersonaScore = (props) => {
+  const { resultsScore, personaName } = props;
+  const {
+    colour,
+    hierarchicalCategoriesLvl0,
+    hierarchicalCategoriesLvl1,
+    hierarchicalCategoriesLvl2,
+    hierarchicalCategoriesLvl3,
+    genderFilter,
+    category,
+    brand,
+    categories,
+  } = hitsConfig;
+  const [colourArray, setColourArray] = useState([]);
```

Comment:
We might have the possibility to create one state that contain an object : 

`const [setObject, object] = useState({
color: [],
brand: [],
... etc
})` 

Maybe I'm wrong 😊

Reply:
Yes I think you're right it's just with the step to get data from the array if it's not empty push the new value to existant values.
I don't know if we can do it:
```suggestion
          if (colourArray.length > 0) {
          setColourArray([...colourArray, resultsScore.colour]);
        }
```